### PR TITLE
AB#99726 Added frontdoor decision persistance

### PR DESCRIPTION
**Pull request by MichalSzrederPWC**
### 🛠 Changes being made

Added saving eligibility result on frontdoor project submit. Used hardcoded ints because appropriate enum was not present yet in crm.

### 📸 Screenshots (optional)

Insert some screenshots if you made UI changes.

### 🏎 Quality check

- [x] Have you performed local tests?
- [x] Are integration tests passing locally?
- [ ] Have you added some unit tests?
- [ ] Have you added some integration tests?
- [ ] Have you checked WCAG (included screenshot)

### CODE REVIEW

File: src/HE.Investments.FrontDoor.Domain/Project/ProjectEntity.cs [hunk 1]
```diff
@@ -35,7 +35,8 @@ public class ProjectEntity : DomainEntity
         RequiredFunding? requiredFunding = null,
         IsProfit? isProfit = null,
         ExpectedStartDate? expectedStartDate = null,
-        ProjectLocalAuthority? localAuthority = null)
+        ProjectLocalAuthority? localAuthority = null,
+        int? frontDoorDecision = null)
```
Comment (jjankowski87) [inline src/HE.Investments.FrontDoor.Domain/Project/ProjectEntity.cs:39]: Use enum in domain entity, map to DTO on repository level

File: src/HE.Investments.FrontDoor.WWW/Controllers/ProjectController.cs [hunk 2]
```diff
@@ -516,6 +516,7 @@ public async Task<IActionResult> CheckAnswers(CancellationToken cancellationToke
     public async Task<IActionResult> Complete([FromRoute] string projectId, CancellationToken cancellationToken)
     {
         var (operationResult, applicationType) = await _mediator.Send(new ValidateProjectAnswersQuery(FrontDoorProjectId.From(projectId)), cancellationToken);
+        await _mediator.Send(new UpdateFrontDoorDecisionCommand(FrontDoorProjectId.From(projectId), applicationType), cancellationToken);
```
Comment (jjankowski87) [inline src/HE.Investments.FrontDoor.WWW/Controllers/ProjectController.cs:519]: I belive this call should be below if, so decision is saved when there is no error in answers

**Review by Carq — CHANGES_REQUESTED** [review]
As we spoke the whole logic related with saving decision should be inside ProjectEntity

File: src/HE.Investments.FrontDoor.WWW/Controllers/ProjectController.cs (hunk 2 above)
Comment (MichalSzrederPWC) [reply src/HE.Investments.FrontDoor.WWW/Controllers/ProjectController.cs]: moved logic to projectentity

File: src/HE.Investments.FrontDoor.Domain/Project/ProjectEntity.cs (hunk 1 above)
Comment (MichalSzrederPWC) [reply src/HE.Investments.FrontDoor.Domain/Project/ProjectEntity.cs]: done

**Comment by MichalSzrederPWC** [comment]
> As we spoke the whole logic related with saving decision should be inside ProjectEntity

done

**Review by jjankowski87 — APPROVED** [review]

**Review by Carq — APPROVED** [review]

**Pull request merged**